src/components/AudioPlayer/AudioPlayer.js

```diff
@@ -198,14 +198,25 @@ class AudioPlayerSub extends Component {
 
   // When props change
   componentDidUpdate(prevProps) {
-    const { playing } = this.props
+    // const { playing } = this.props
+    const { playing, progress, updatePlaying, topScreen } = this.props
+
+    // If song has ended, reset progress and trigger end action
+    if (topScreen && Math.round(progress * 10000) / 10000 >= 1) {
+      const { updatePlayed, updateProgress, endSong } = this.props
+
+      updatePlayed(0)
+      updateProgress(0)
+
+      endSong()
+      updatePlaying(false)
+    }
```

Comment:
I believe this same (or very similar) logic exists in `ProgressBar`, do you have any ideas why it needs to be here for Android? I think we probably need to either move it here entirely or check the platform and only execute this code on Android.